plugins/none-ls/_mk-source-plugin.nix

```diff
@@ -0,0 +1,50 @@
+# mkSourcePlugin, returns a module
+sourceType: sourceName:
+{
+  config,
+  pkgs,
+  lib,
+  helpers,
+  ...
+}:
+let
+  inherit (import ./packages.nix pkgs) packaged;
+  pkg = packaged.${sourceName};
+
+  cfg = config.plugins.none-ls;
+  cfg' = config.plugins.none-ls.sources.${sourceType}.${sourceName};
+in
+{
+  options.plugins.none-ls.sources.${sourceType}.${sourceName} =
+    {
+      enable = lib.mkEnableOption "the ${sourceName} ${sourceType} source for none-ls";
+      withArgs = helpers.mkNullOrOption helpers.nixvimTypes.strLua ''
+        Raw Lua code passed as an argument to the source's `with` method.
+      '';
+    }
+    # Only declare a package option if a package is required
+    // lib.optionalAttrs (packaged ? ${sourceName}) {
+      package = lib.mkOption (
+        {
+          type = lib.types.nullOr lib.types.package;
+          description =
+            "Package to use for ${sourceName} by none-ls. "
+            + (lib.optionalString (pkg == null) ''
+              Not handled in nixvim, either install externally and set to null or set the option with a derivation.
```

Comment:
maybe in `nixpkgs` instead of `nixvim`? As we try to be up-to-date. Though this means that we sort-of garantee to check the unpackaged sources, I'm not sure if we want that.
In the end, as you wish I think

Reply:
I just copy-pasted the existing description. I agree we can improve the wording.

Maybe:

> Currently not packaged in nixpkgs.
> Either set this to `null` and install ${sourceName} outside of nix, or set this to a custom or third-party nix package.

Reply:
Looks good!

Reply:
Done